### DISCUSSION TITLE
How to convert a 3D vector to screen (or view) pixel coordinate system

### DISCUSSION THREAD

**Discussion by guiguite** [google/filament#7101] [Q&A]
How to get the central point of an entity or asset and convert it into the 2D screen or view coordinate system?
(ios / android)

**Comment by sebapian** [comment]
Looking to do something similar too, upvote!

**Comment by jeanlemotan** [comment]
There was a similar question some time ago, but they wanted to to the inverse - convert a screen point to a 3D point:
https://github.com/google/filament/discussions/6906

You want to project a 3D point to a screen coordinate. This is done by simply multiplying the 3D point with the viewProjection matrix. This will give you a point in normalized coordinates (-1 .. 1 on all axis). If the result is lower than -1 or bigger than 1 on any axis, then it's out of screen (actually out of the frustum). You then center and scale the result to convert to pixel coords.

**Reply by guiguite** [reply]
Thanks! Will try and update this discussion once done.
